Foundation/JSONSerialization.swift

```diff
@@ -285,219 +454,176 @@ internal extension JSONSerialization {
 }
 
 //MARK: - JSONSerializer
-private struct JSONWriter {
+private struct JSONWriter : JSONVisitor {
 
     var indent = 0
     let pretty: Bool
     let sortedKeys: Bool
-    let writer: (String?) -> Void
+    var result = ""
```

Comment:
Make the optimizer's job a little easier. Nothing was using this flexibility currently.